utils/username.py

```diff
@@ -57,11 +57,45 @@ def redirect_if_old_username_or_404(func):
     """
 
     def inner(request, *args, **kwargs):
-        new_user = get_user_or_404(kwargs['username'])
-        if kwargs['username'] != new_user.username:
-            kwargs['username'] = new_user.username
+        if hasattr(request, 'parameter_user'):
+            # If request.parameter_user already exists because it was added by some other decorator, reuse it
+            user = request.parameter_user
+        else:
+            # Otherwise get the corresponding user (considering OldUsernames) object or raise 404
+            user = get_user_or_404(kwargs['username'])
+
+        if kwargs['username'] != user.username:
+            # If the the username is an old username of the user, do redirect
+            kwargs['username'] = user.username
             return redirect(reverse(inner, args=args, kwargs=kwargs), permanent=True)
-        request.parameter_user = new_user
+
+        # Save user object in the request so it can be used by the view and/or other decorators
+        request.parameter_user = user
+        return func(request, *args, **kwargs)
+
+    return inner
+
+
+def raise_404_if_user_is_deleted(func):
```

Comment:
It looks like we have to apply these two decorators in the correct order, and on very specific views. I wonder if it makes sense to start a development guide that says things like this. For example, this could be in a section called "URLs that include a username" or something

Reply:
Good idea, added documentation 56df8a8d211b99c0bf2b16708dea75f6be518aa9